src/lib/CustomGraphiQL.js

```diff
@@ -0,0 +1,39 @@
+// JS is used here instead of TS because of a type issue in GraphiQL.
+// The 'schema' prop is documented to accept GraphQLSchema | null | undefined
```

Comment:
Okayy :thinking: . I would have rather hacked around this exception by using `as` and continued with TypeScript. But this is okay too.
Can we at least have proptypes for this component?

Reply:
`schema === null ? undefined: schema` will not work because, like I've described in the comment, we need to pass `null` to avoid the GraphiQL component from sending an introspection query. Apologies if this is not clear from the comment.

`as` would have worked but to be honest, I never knew about the `as unknown as someType` trick to coerce any type into anything else.

Anyways, I fixed this in the graphiql repo and my PR https://github.com/graphql/graphiql/pull/2029 has been merged and included in the latest release. So, I can move this back to TS without any hack.